generate_openj9_scc.sh

```diff
@@ -0,0 +1,295 @@
+
+        # Removing the tar file
+        rm -rf "$DOWNLOAD_PATH_ECLIPSE"
+    fi
+
+    # check for tomcat
+    if [ "$RUN_TOMCAT" == true ]; then
+        # Creating a temporary directory for tomcat download
+        mkdir -p "$DOWNLOAD_PATH_TOMCAT" "$INSTALL_PATH_TOMCAT"
+
+        # Downloading tomcat
+        wget --directory-prefix="$DOWNLOAD_PATH_TOMCAT" https://downloads.apache.org/tomcat/tomcat-9/v9.0.35/bin/apache-tomcat-9.0.35.tar.gz
+
+        # Extracting tomcat
+        tar -xvzf "$DOWNLOAD_PATH_TOMCAT"/apache-tomcat-9.0.35.tar.gz -C "$INSTALL_PATH_TOMCAT"
+
+        # Removing the tar file
+        rm -rf "$DOWNLOAD_PATH_TOMCAT"
+    fi
+
+}
+
+# dry run to right size the cache
+function dry_run() {
+    # Creating base layer first instead of running sample programs to generate SCC directly
+    java -Xshareclasses:name=dry_run_scc,cacheDir=/opt/java/.scc,bootClassesOnly,nonFatal,createLayer -Xscmx$SCC_SIZE -version
+
+    # Pointing cache for sample program runs
+    export OPENJ9_JAVA_OPTIONS="-Xshareclasses:name=dry_run_scc,cacheDir=/opt/java/.scc,bootClassesOnly,nonFatal"
+
+    # check if `xvfb` installed
+    if [ "$INSTALL_XVFB" == true ]; then
+        # Spawning a virtual screen for GUI apps
+        Xvfb :1 -ac -screen 0 1024x768x8 &
+        # Saving the PID
+        XVFB_PID=$!
+        # Saving older display value
+        OLD_DISPLAY=$DISPLAY
+        # Setting DISPLAY to created screen
+        export DISPLAY=:1
+    fi
+
+    for ((i=0; i<SCC_GEN_RUNS_COUNT; i++))
+    do
+        if [ "$RUN_ECLIPSE" == true ]; then
+            run_eclipse_and_stop
+        fi
+        if [ "$RUN_TOMCAT" == true ]; then
+            run_tomcat_and_stop
+        fi
+    done
+
+    # check if `xvfb` installed
+    if [ "$INSTALL_XVFB" == true ]; then
+        # Killing `xvfb` process to remove screen
+        kill -9 $XVFB_PID
+        # Setting back the DISPLAY to older value
+        export DISPLAY=$OLD_DISPLAY
+    fi
+
+
+    FULL=$( (java -Xshareclasses:name=dry_run_scc,cacheDir=/opt/java/.scc,printallStats || true) 2>&1 | awk '/^Cache is [0-9.]*% .*full/ {print substr($3, 1, length($3)-1)}')
+
+    java -Xshareclasses:name=dry_run_scc,cacheDir=/opt/java/.scc,destroy || true
+
+    SCC_SIZE="${SCC_SIZE:0:-1}"
+
+    SCC_SIZE=$(awk "BEGIN {print int($SCC_SIZE * $FULL / 100.0)}")
+
+    [ "$SCC_SIZE" -eq 0 ] && SCC_SIZE=1
+
+    SCC_SIZE="${SCC_SIZE}m"
+
+    # Re-generating cache with new size
+    java -Xshareclasses:name=openj9_system_scc,cacheDir=/opt/java/.scc,bootClassesOnly,nonFatal,createLayer -Xscmx$SCC_SIZE -version
+
+    unset OPENJ9_JAVA_OPTIONS
+}
+
+# Generate SCC by running apps
+function generate_scc() {
+    # Pointing cache for sample app runs
+    export OPENJ9_JAVA_OPTIONS="-Xshareclasses:name=openj9_system_scc,cacheDir=/opt/java/.scc,bootClassesOnly,nonFatal"
+
+    # check if `xvfb` installed
+    if [ "$INSTALL_XVFB" == true ]; then
```

Comment:
Should be able to re-use function from above.

Reply:
made changes accordingly